arctic/store/version_store.py

```diff
@@ -133,6 +133,9 @@ def list_symbols(self, all_symbols=False, snapshot=None, regex=None, **kwargs):
             query['symbol'] = {'$regex': regex}
         if kwargs:
             for k, v in six.iteritems(kwargs):
+                # TODO: this doesn't work as expected as it ignores the versions with metadata.deleted set
+                #       as a result it will return symbols with matching metadata which have been deleted
+                #       Maybe better add a match step in the pipeline instead of making it part of the query
```

Comment:
if this doesnt work, should we remove it?

Reply:
We had a brief chat on this.

Seems that apart from some integration tests, users don't typically pass metadata args with list_symbols. Is this a valid assertion?

We can either remove this functionality and change the api/behaviour with this PR, or fix it with another one.
IMHO the fix should only touch/modify the pipeline if metadata arguments are passed, i.e. remain unmodified as-is if no metadata args are present. An extra $match stage could do.